api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreServiceImpl.java

```diff
@@ -432,19 +450,34 @@ public BiometricData saveBiometricsForPatient(Patient patient, BiometricData bio
 				}
 				if (identifierExists) {
 					log.debug("Identifier already exists for patient");
-				}
-				else {
-					PatientIdentifier identifier = new PatientIdentifier(subject.getSubjectId(), idType, null);
+				} else {
+					PatientIdentifier identifier = identifierBuilder.createIdentifier(idType.getUuid(), subject.getSubjectId(), null);
 					patient.addIdentifier(identifier);
 					patientService.savePatientIdentifier(identifier);
 					log.debug("New patient identifier saved for patient: " + identifier);
 				}
-			}
-			else {
+			} else {
 				// TODO: In the future we could add additional support for different storage options - eg. as person attributes
 				throw new IllegalArgumentException("Invalid biometric configuration.  No patient identifier type specified");
 			}
 		}
 		return biometricData;
 	}
+
+	@Override
+	public BiometricData generateBiometricData(BiometricSubject biometricSubject) {
```

Comment:
I will remove it. I created that beacuse I thought that it could be also used in other places.